### PR TITLE
Resolve confusing source IDs in web interfaces

### LINKED ISSUE #1998: Resolve inconsistently named and confusing source IDs
## Description

When working with the SecureDrop application code, there are multiple IDs referred to as the `source_id`:

* the filesystem ID (the string used to refer to sources on disk)
  * e.g. referred to as `source_id` in `delete_collection()` in `journalist.py`
* the source ID (primary key of the source in the SecureDrop database)
  * e.g. referred to as `source_id` in `index()` in `journalist.py`
* the `sid` - it is unclear what this is on first inspection (it's the filesystem ID but these instances of `sid` should be replaced with which column this is referring to)

These are inconsistently named across the application and this is very confusing for new developers. 

## Fix

We should resolve this confusion by using these variables in a consistent manner.

There should be two IDs corresponding to the two columns in the `sources` table:

* `filesystem_id`: replace instances of `sid` and replace any `source_id`s that actually refer to the `filesystem_id`
* `source_id`: this should _always_ refer to the primary key of the source in the `sources` table

### PR DESCRIPTION
**Pull request by redshiftzero**
## Status

Ready for review

## Description of Changes

Fixes #1998 (a Flask webapp cleanup issue)

Changes proposed in this pull request:

 * Remove `sid` variable and replace with `filesystem_id`
 * Replace instances of `source_id` that should actually be `filesystem_id`

## Testing

Best to inspect the webapp code to verify that my substitutions are correct. 

CI tests should pass.

## Deployment

Should not have any deployment issues.

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

### CODE REVIEW

**Comment by dachary** [comment]
Note to self: because the modified files do not have 100% code coverage, it would be good to verify that each modified line is covered, to avoid a typo hidden in plain sight.

**Comment by dachary** [comment]
verified that only the substitutions explained in the description are in the diff

**Comment by dachary** [comment]
@redshiftzero needs rebase

**Comment by redshiftzero** [comment]
Rebased on latest develop

**Comment by dachary** [comment]
Any idea why the two circleci checks are still pending ? (docs-lint, staging-test)

**Comment by redshiftzero** [comment]
Ah, good question - that's because we made a change to CI in PR #2149 to not run staging tests for documentation PRs. When we merged that PR in, [we flipped the required checks](https://github.com/freedomofpress/securedrop/pull/2149#issuecomment-323466190) from `ci/circleci` to `ci/circleci: docs-lint` and `ci/circleci: staging-test`, but since this PR was submitted before that change was made, the two required checks are waiting for status to be reported. Since it's easy to do so, I went ahead and repushed in this case :smile:

File: securedrop/journalist_templates/col.html [hunk 1]
```diff
@@ -94,9 +95,9 @@
   <hr class="cut-out">
   <p>Click below to delete this source's collection. <em>Warning: If you do this, the files seen here will be unrecoverable and the source will no longer be able to login using their previous codename.</em></p>
 
-  <form id="delete-collection" action="/col/delete/{{sid}}" method="post">
+  <form id="delete-collection" action="/col/delete/{{filesystem_id}}" method="post">
     <input name="csrf_token" type="hidden" value="{{ csrf_token() }}">
-    <input type="hidden" name="sid" value="{{ sid }}">
+    <input type="hidden" name="sid" value="{{ filesystem_id }}">
```
Comment (heartsucker) [inline securedrop/journalist_templates/col.html]: `name="sid"` should be changed.

**Comment by redshiftzero** [comment]
Recall that `source_id` is the primary key of the source in the SecureDrop database (`1`, `2`, etc.), and `filesystem_id` is the string used to refer to the source on disk (`5XJTHIB47VYS72OVFBBUATNWDGQDXZCHAN3WLVOS4YO2JMXODNLRNM4PY5ZNUEZF52LTQS2JYXNDZGECPHOU6RRGMQXTQF5TUJGUV5Q=`, `TAK74LXCGSUWESNLEYRMQXSNLMW4CNLRQZLBJXH3SNGC75XICQATSNR2DUMHRW65UXE3TIYYOD7U5PBSATRRD5J6ZE6Z42RNYGFPZXA=`, etc.)

**Comment by heartsucker** [comment]
Ah yes. Thanks for the reminder.

File: securedrop/journalist_templates/col.html (hunk 1 above)
Comment (redshiftzero) [inline securedrop/journalist_templates/col.html]: Good call, fixed!

**Comment by conorsch** [comment]
Not merging since I see @dachary is flagged for review. For the record, I'm OK with merge if both @redshiftzero and @heartsucker say ✅  (which they have).

**Comment by redshiftzero** [comment]
Thanks for being explicit on that @conorsch - I'm fine with this going in (I pinged both @heartsucker and @dachary for whoever got to it first) so merging!

**Pull request merged**